src/core/etl/src/Flow/ETL/DataFrame.php

```diff
@@ -634,6 +634,16 @@ public function renameAll(string $search, string $replace) : self
         return $this;
     }
 
+    /**
+     * @lazy
+     */
+    public function renameAllCallback(\Closure $callback) : self
```

Comment:
I'm not a fan of adding more callbacks to data frame. I started working on a parallel processing on a side and each api method that adds callback is just a pure hell since in parallel processing I need to serialize and spread "transformations" among many processes which with callbacks is extremely unstable.

Even for map I had to put an exception that DataFrame::map() cant be used with parallel processing and that its recommended to implement a scalar function or transformation instead.